Entities/AHuman.cpp

```diff
@@ -3460,13 +3462,19 @@ void AHuman::Update()
 				heldDevice->SetAngularVel(heldDevice->GetAngularVel() + m_AngularVel * 0.5F + 3.0F * RandomNormalNum());
 
 				arm->SetHandPos(heldDevice->GetPos());
-				if (!m_Inventory.empty()) {
-					arm->SetHeldDevice(dynamic_cast<HeldDevice *>(SwapNextInventory()));
-					arm->SetHandPos(m_Pos + RotateOffset(m_HolsterOffset));
+
+                HeldDevice* BGArmDevice = m_pBGArm->GetHeldDevice();
```

Comment:
Calling `GetHeldDevice()` on `BGArm` like this without first checking whether the arm exists will result in the game crashing if the arm isn't there!